modules/rostests/apitests/compiler/floatconv.c

```diff
@@ -0,0 +1,278 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Test for floating point conversion
+ * COPYRIGHT:   Copyright 2024 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+#include <apitest.h>
+
+/* Note: There are 2 behaviors for float to unsigned long conversion:
+ * 1. The old behavior, which exists only on x86 and CL versions up to somewhere
+ *    between 19.40.33811 and 19.41.33923:
+ *     - If a negative float is cast to an unsigned integer, the result is ULONG_MAX
+ *       for uint32 and ULLONG_MAX for uint64.
+ *     - If a float is cast to an unsigned long and the value is larger than ULONG_MAX,
+ *       the result is ULONG_MAX
+ * 2. The new behavior (x64 and on x86 versions after the ones mentioned above):
+ *     - If a negative float is cast to an unsigned integer, the result is the same as
+         first casting to the signed type, then casting to the unsigned type.
+ *     - If a float is cast to an unsigned integer and the value is too large for the type,
+ *       the result is 0 for uint32 and 0x8000000000000000 for uint64.
+ * In the old version, the float to unsigned conversion was inlined and used _ftoul2 and
+ * a comparison against 0x43e0000000000000 (9223372036854775808.0) to check for overflow.
+ * In the new version, the float to unsigned conversion is done by a call to _ftoul2_legacy,
+ * which checks for overflow by comparing against 0x5f800000 (18446744073709551616.0) and
+ * then forwards the call to either _ftol2 or _ftoul2.
+ */
+
+#if defined(_M_IX86) && defined(__VS_PROJECT__) && (_MSC_FULL_VER < 194133923)
```

Comment:
Because only the VS project links to the implementation shipped with VS, which is responsible for this behavior. Our implementation does not reproduce that behavior.

Reply:
understandable